### PR TITLE
Inlier residuals checkbox

### PR DESCRIPTION
**Pull request by mleotta**
Added a check box to show only inlier residuals.  Previously this was the default behavior.  Now unchecking the box shows outliers too.

### CODE REVIEW

File: gui/CameraView.cxx [hunk 1]
```diff
@@ -103,18 +105,24 @@ class ActorColorOption : public QWidget
...
-  : QWidget(parent), button(new ActorColorButton(this))
+  : QWidget(parent), button(new ActorColorButton(this)),
+    inlierCheckbox(new QCheckBox(this))
 {
   auto const layout = new QFormLayout(this);
   layout->addRow("Color", this->button);
+  layout->addRow("Inliers Only", this->inlierCheckbox);
 
   this->button->persist(this->uiState, settingsGroup + "/Color");
+  auto const cboxItem = new qtUiState::Item<Qt::CheckState, QCheckBox>(
```
Comment (mwoehlke-kitware) [inline gui/CameraView.cxx]: `mapChecked`? (Note: `QCheckBox` subclasses `QAbstractButton`.)

File: gui/CameraView.cxx [hunk 2]
```diff
@@ -208,11 +216,13 @@ class CameraViewPrivate
   vtkNew<vtkMatrix4x4> transformMatrix;
 
   LandmarkCloud landmarks;
-  SegmentCloud residuals;
+  SegmentCloud residuals_inlier;
```
Comment (mwoehlke-kitware) [inline gui/CameraView.cxx]: BTW, inconsistent naming convention (s.b. e.g. `residualsInlier`).

File: gui/CameraView.cxx [hunk 3]
```diff
@@ -666,13 +683,21 @@ void CameraView::addLandmark(
...
+  bool inlier)
 {
   QTE_D();
 
   Q_UNUSED(id)
 
-  d->residuals.addSegment(x1, y1, -0.2, x2, y2, -0.2);
+  if (inlier)
+  {
+    d->residuals_inlier.addSegment(x1, y1, -0.2, x2, y2, -0.2);
```
Comment (mwoehlke-kitware) [inline gui/CameraView.cxx]: Would there be any benefit assigning these different (from each other) z-orders? (The answer may be "no", but something to think about if you hadn't already.)

File: gui/CameraView.cxx [hunk 4]
```diff
@@ -720,7 +746,19 @@ void CameraView::setResidualsVisible(bool state)
...
+    !d->residualsOptions->inlierCheckbox->isChecked());
+  this->render();
+}
+
+//-----------------------------------------------------------------------------
+void CameraView::setOutlierResidualsVisible(bool state)
+{
+  QTE_D();
+
+  bool overall_state = d->residuals_inlier.actor->GetVisibility();
```
Comment (mwoehlke-kitware) [inline gui/CameraView.cxx]: BTW, naming here also; the Qt-ish code is mostly `thisCase`, not `this_case`.

**Comment by mleotta** [comment]
@mwoehlke-kitware I've made all of your suggested changes.

**Pull request merged**